java/hiro-client/src/main/java/co/arago/hiro/client/rest/DefaultWebSocketClient.java

```diff
@@ -257,6 +347,62 @@ public synchronized void sendMessage(String message) {
         }
     }
 
+    @Override
+    public synchronized void addEventFilter(Map filter) {
+        String id = getFilterId(filter);
+        String message = getEventRegisterMessage(filter);
+        LOG.log(Level.INFO, "Add filter: " + message);
+        sendMessage(message);
+        eventFilterMessages.put(id, filter);
+    }
+
+    private String getFilterId(Map filter) {
+        String id = String.valueOf(filter.get("filter-id"));
+        if (id == null || StringUtils.isEmpty(id)) {
+            throw new HiroException("Wrong filter specification. Key 'filter-id' is missing.", 400);
+        }
+        return id;
+    }
+
+    private String getEventRegisterMessage(Map filter) {
+        final Map m = HiroCollections.newMap();
+        m.put("type", "register");
+        m.put("args", filter);
+
+        String message = JSONValue.toJSONString(m);
+
+        return message;
+    }
+
+    @Override
+    public synchronized void removeEventFilter(String id) {
+        final Map m = HiroCollections.newMap();
+        m.put("type", "unregister");
+        m.put("args", HiroCollections.newMap("filter-id", id));
+
+        String message = JSONValue.toJSONString(m);
+
+        LOG.log(Level.INFO, "Remove filter: " + message);
+
+        sendMessage(message);
+        eventFilterMessages.remove(id);
```

Comment:
maybe make it concurrent collection to avoid client multithreaded access problems?

Reply:
Sure can do.